src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -3649,6 +3653,23 @@ public List<GHRepositoryTrafficTopReferralSources> getTopReferralSources() throw
                 .fetch(GHRepositoryTrafficTopReferralSources[].class));
     }
 
+    /**
+     * Get all active rules that apply to the specified branch
+     * (https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-rules-for-a-branch).
+     *
+     * @param branch
+     *            the branch
+     * @return the rules for branch
+     * @throws IOException
+     *             the io exception
+     */
+    public List<GHRepositoryRule> getRulesForBranch(String branch) throws IOException {
```

Comment:
This endpoint uses pagination.  
```suggestion
    public PagedIterable<GHRepositoryRule> listRulesForBranch(String branch) throws IOException {
```